torch/quantization/fake_quantize.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+from torch.nn import Module
+from .observer import default_observer
+
+class FakeQuantize(Module):
+    ''' Simulate the quantize and dequantize operations in training time.
+    Args:
+        `qconfig`: object that encodes configuration info for quantization
+        `observer_module`: Observer module that records stats of weights and
+        activations
+        `calcqparam`: A function that calculates quantization parameters
+        given the stats
+    '''
+
+    def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
+                 quant_min=0, quant_max=255, enable_fq=True):
+        super(FakeQuantize, self).__init__()
+        assert torch.iinfo(dtype).min <= quant_min, 'quant_min out of bound'
+        assert quant_min <= quant_max, \
+            'quant_min must be less than or equal to quant_max'
+        assert quant_max <= torch.iinfo(dtype).max, 'quant_max out of bound'
+        self.dtype = dtype
+        self.qscheme = qscheme
+        self.quant_min = quant_min
+        self.quant_max = quant_max
+        self.enable_fq = enable_fq
+        self.observer = default_observer(dtype=dtype, qscheme=qscheme)()
+        self.scale = None
+        self.zero_point = None
+
+    def enable(self):
```

Comment:
and what about disable? :)

maybe have it be a property, just like module.training? maybe just self.enabled